From 1f62dd92bf54467bc8122ae3deb4d3b33c42814b Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 14 Jan 2016 09:59:52 +0800
Subject: [PATCH 1420/1691] MLK-12251 video: mipi_dsi: fix an unsigned compared
 against zero issue

commit 1f62dd92bf54467bc8122ae3deb4d3b33c42814b from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

Compared 'mux' against zero is meaningless. So correct this.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
---
 drivers/video/fbdev/mxc/mipi_dsi.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/fbdev/mxc/mipi_dsi.c b/drivers/video/fbdev/mxc/mipi_dsi.c
index 3ebf25b..3e78a53 100644
--- a/drivers/video/fbdev/mxc/mipi_dsi.c
+++ b/drivers/video/fbdev/mxc/mipi_dsi.c
@@ -790,7 +790,7 @@ static int mipi_dsi_probe(struct platform_device *pdev)
 	struct resource *res;
 	u32 dev_id, disp_id;
 	const char *lcd_panel;
-	unsigned int mux;
+	int mux;
 	int ret = 0;
 
 	mipi_dsi = devm_kzalloc(&pdev->dev, sizeof(*mipi_dsi), GFP_KERNEL);
@@ -889,7 +889,7 @@ static int mipi_dsi_probe(struct platform_device *pdev)
 	mux = mipi_dsi->bus_mux->get_mux(dev_id, disp_id);
 	if (mux >= 0)
 		regmap_update_bits(mipi_dsi->regmap, mipi_dsi->bus_mux->reg,
-				   mipi_dsi->bus_mux->mask, mux);
+				   mipi_dsi->bus_mux->mask, (unsigned int)mux);
 	else
 		dev_warn(&pdev->dev, "invalid dev_id or disp_id muxing\n");
 
-- 
1.9.1

